### PR TITLE
Test: Configure performance tests for math libraries

### PR DESCRIPTION
**Pull request by jieli-matrix**
### Description

Configure performance tests for math libraries by using [google benchmark](https://github.com/google/benchmark.git) as I mentioned in #3491.

A good start for Issue #3550.

ToDo:
- [x] Configure benchmark in cmake.
- [x] Configure workflows.
- [x] Add benchmark test for `math_sphbes`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -608,6 +610,33 @@ IF (BUILD_TESTING)
   endfunction(AddTest)
 endif()
 
+
+#  Add performance test in abacus
+IF (ENABLE_GOOGLEBENCH)
+  option(BUILD_TESTING "GoogleBenchmark relies on google googletest toolkit." ON)
```
Comment (caic99) [inline CMakeLists.txt]: Please put this option ahead of the setup of googletest.

File: CMakeLists.txt [hunk 2]
```diff
@@ -608,6 +610,33 @@ IF (BUILD_TESTING)
...
+  find_package(benchmark HINTS /usr/local/lib/ ${BENCHMARK_DIR})
```
Comment (caic99) [inline CMakeLists.txt]:
```suggestion
  find_package(benchmark HINTS ${BENCHMARK_DIR})
```

**Review by caic99 — CHANGES_REQUESTED** [review]

File: Dockerfile.gnu [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FROM ubuntu:22.04
 RUN apt update && apt install -y --no-install-recommends \
     libopenblas-openmp-dev liblapack-dev libscalapack-mpi-dev libelpa-dev libfftw3-dev libcereal-dev \
-    libxc-dev libgtest-dev libgmock-dev python3-numpy \
+    libxc-dev libgtest-dev libgmock-dev libbenchmark-dev python3-numpy \
```
Comment (caic99) [inline Dockerfile.gnu:4]: Please also update docs. Thanks!

**Review by caic99 — APPROVED** [review]

File: Dockerfile.gnu (hunk 3 above)
Comment (jieli-matrix) [reply Dockerfile.gnu:4]: Done!

**Comment by caic99** [comment]
@jieli-matrix I suggest putting benchmark test in the existed integrated test process, avoid overheads. Maybe in some day we have enough tests, we could separate them into a workflow, and run routinely.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ jobs:
 
       - name: Build
         run: |
-          cmake -B build -DBUILD_TESTING=ON -DENABLE_DEEPKS=ON -DENABLE_LIBXC=ON -DENABLE_LIBRI=ON -DENABLE_PAW=ON
+          cmake -B build -DENABLE_GOOGLEBENCH=ON -DENABLE_DEEPKS=ON -DENABLE_LIBXC=ON -DENABLE_LIBRI=ON -DENABLE_PAW=ON
```
Comment (caic99) [inline .github/workflows/test.yml]:
```suggestion
          cmake -B build -DBUILD_TESTING=ON -DENABLE_DEEPKS=ON -DENABLE_LIBXC=ON -DENABLE_LIBRI=ON -DENABLE_PAW=ON -DENABLE_GOOGLEBENCH=ON
```

**Review by caic99 — APPROVED** [review]
LGTM

**Pull request merged**